### PR TITLE
Week 7 js apis and php oop

### PR DESCRIPTION
**Pull request by billythakidd04**
Added Student and Group classes using OOP for php to build and utilize a form. It's very sketchy and needs work but the principle functionality exists.

### CODE REVIEW

**Review by gc824 — APPROVED** [review]

**Review by vricci518 — APPROVED** [review]

**Review by eiman-kased — APPROVED** [review]

**Review by CharleswithLove — COMMENTED** [review]
I need more guidance on how to resolve the two conflicting files if they havent been

**Comment by billythakidd04** [comment]
> I need more guidance on how to resolve the two conflicting files if they havent been

They are not yours to resolve. The person who creates the branch/pr is responsible for fixing conflicts.

**Review by ghost — APPROVED** [review]

File: Week_7_OOP_JSAPIs/src/Student/student.php [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	private $email;
+	private $repositoryURL;
+	private $groupID;
+
+	private static $db;
+
+	private static function getDBConn()
+	{
+		// connect to db
+		self::$db = self::$db ?? dbConn();
```
Comment (nburd90) [inline Week_7_OOP_JSAPIs/src/Student/student.php:18]: Can we go through this line of code?

File: Week_7_OOP_JSAPIs/src/Student/student.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	/**
+	 * Undocumented function
+	 *
+	 * @return boolean
+	 */
+	public function createStudent(): bool
+	{
+		self::getDBConn();
+
+		$f = self::$db->real_escape_string($this->firstName);
```
Comment (nburd90) [inline Week_7_OOP_JSAPIs/src/Student/student.php:80]: What if it's not a string??

**Review by nburd90 — APPROVED** [review]

**Review by margeschrec — APPROVED** [review]

**Review by WaymonBrown84 — APPROVED** [review]

**Review by nburd90 — APPROVED** [review]

**Review by CharleswithLove — APPROVED** [review]

**Pull request merged**